src/libs/ajax/data-table-providers/WdsDataTableProvider.ts

```diff
@@ -21,7 +22,7 @@ import { notifyDataImportProgress } from 'src/workspace-data/import-jobs';
 // interface definitions for WDS payload responses
 export interface AttributeSchema {
   name: string;
-  datatype: string;
+  datatype: string | null;
   relatesTo?: string;
```

Comment:
I added a null here so I didn't have to specify it in order to rename the column.  Should it be optional instead?  Is there a better solution?

Reply:
My vote would be to add a different, more specific type for the update attribute request body. Something like:
```ts
type AttributeSchemaUpdate = { name: string } | { datatype: string };
```

We could also define a union type for data types and make this something like:
```ts
type AttributeSchemaUpdate = { name: string } | { datatype: AttributeDataType };
```

Reply:
But keep the `AttributeSchema` and also add in an `AttributeSchemaUpdate`?  Seems a bit redundant.

Reply:
We had a similar debate on the Java side. The tradeoff is either loosen the AttributeSchema type to allow it to be used for updates or add another, similar, type. I much prefer to have types as strict as possible to get the most help from TypeScript.

Reply:
Ok well I've added in `AttributeSchemaUpdate`